multicluster/controllers/multicluster/member/labelidentity_controller.go

```diff
@@ -135,12 +143,43 @@ func (r *LabelIdentityReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Watches(&source.Kind{Type: &v1.Namespace{}},
 			handler.EnqueueRequestsFromMapFunc(r.namespaceMapFunc),
 			builder.WithPredicates(predicate.LabelChangedPredicate{})).
+		Watches(&source.Kind{Type: &mcv1alpha2.ClusterSet{}},
+			handler.EnqueueRequestsFromMapFunc(r.clusterSetMapFunc),
+			builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		WithOptions(controller.Options{
 			MaxConcurrentReconciles: common.LabelIdentityWorkerCount,
 		}).
 		Complete(r)
 }
 
+func (r *LabelIdentityReconciler) clusterSetMapFunc(a client.Object) []reconcile.Request {
+	clientSet := &mcv1alpha2.ClusterSet{}
+	requests := []reconcile.Request{}
+	if a.GetNamespace() != r.namespace {
+		return requests
+	}
+	ctx := context.TODO()
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: a.GetNamespace(), Name: a.GetName()}, clientSet)
+	if err == nil {
+		podList := &v1.PodList{}
+		r.Client.List(ctx, podList)
+		requests = make([]reconcile.Request, len(podList.Items))
+		r.recreateMutex.Lock()
+		defer r.recreateMutex.Unlock()
+		for i, pod := range podList.Items {
+			podNamespacedName := types.NamespacedName{
+				Name:      pod.GetName(),
+				Namespace: pod.GetNamespace(),
+			}
+			requests[i] = reconcile.Request{
+				NamespacedName: podNamespacedName,
+			}
+			r.podLabelToRecreate.Insert(podNamespacedName.String())
```

Comment:
The cost seems high to process every Pod one by one. Do we have a more efficient way?

Reply:
I think we need to process every Pod when the ClusterSet is recreated. Even we have some cached label data, we can't tell if there is any Pod label update events during the time windows before the ClusterSet is recreated. 
For Pod event handling, we tuned the concurrent go-routine number via this PR https://github.com/antrea-io/antrea/pull/5099 for a better performance.